lambda_function.py

```diff
@@ -0,0 +1,46 @@
+from pypdf import PdfReader, PdfWriter
+import uuid
+import os
+from urllib.parse import unquote_plus
+import boto3
+
+# Create the S3 client to download and upload objects from S3
+s3_client = boto3.client('s3')
+
+def lambda_handler(event, context):
+    # Iterate over the S3 event object and get the key for all uploaded files
+    for record in event['Records']:
+        bucket = record['s3']['bucket']['name']
+        key = unquote_plus(record['s3']['object']['key']) # Decode the S3 object key to remove any URL-encoded characters
+        download_path = f'/tmp/{uuid.uuid4()}.pdf' # Create a path in the Lambda tmp directory to save the file to 
+        upload_path = f'/tmp/converted-{uuid.uuid4()}.pdf' # Create another path to save the encrypted file to
+        
+        # If the file is a PDF, encrypt it and upload it to the destination S3 bucket
+        if key.lower().endswith('.pdf'):
+            s3_client.download_file(bucket, key, download_path)
+            encrypt_pdf(download_path, upload_path)
+            encrypted_key = add_encrypted_suffix(key)
+            s3_client.upload_file(upload_path, f'{bucket}-encrypted', encrypted_key)
+
+# Define the function to encrypt the PDF file with a password
+def encrypt_pdf(file_path, encrypted_file_path):
+    reader = PdfReader(file_path)
+    writer = PdfWriter()
+    
+    for page in reader.pages:
+        writer.add_page(page)
+
+    # Add a password to the new PDF
+    # In this example, the password is hardcoded.
+    # In a production application, don't hardcode passwords or other sensitive information.
+    # We recommend you use AWS Secrets Manager to securely store passwords.
+    writer.encrypt("my-secret-password")

```

Comment:
Try to use your own .env to do so. 

Use a sample .env, and write the fact that it is you that hardcoded it